### PR TITLE
Program group reorganization

### PR DESCRIPTION
**Pull request by cmnbroad**
Reorganize program group categories and tool assignments, per @vdauwera. There are 3 separate commits; one with the new program group class definitions; one with the tool assignments; and one to delete the obsolete program groups.

### Checklist (never delete this)
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [X] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

### CODE REVIEW

File: src/main/java/picard/vcf/MakeSitesOnlyVcf.java [hunk 1]
```diff
@@ -36,7 +37,7 @@
                 "all site level information, including annotations based on genotypes (e.g. AN, AF). Output an be " +
                 "any support variant format including .vcf, .vcf.gz or .bcf.",
         oneLineSummary = "Creates a VCF bereft of genotype information from an input VCF or BCF",
-        programGroup = VcfOrBcf.class)
+        programGroup = VariantEvaluationProgramGroup.class)
```
Comment (sooheelee) [inline src/main/java/picard/vcf/MakeSitesOnlyVcf.java]: programGroup = VariantManipulationProgramGroup.class

**Comment by sooheelee** [comment]
I am good with these. @vdauwera Will you want to review too?

**Comment by vdauwera** [comment]
Ah, I had already looked at it but failed to post my review. All looks good to me!

**Comment by sooheelee** [comment]
👍🏽

**Comment by ktibbett** [comment]
I followed the directions for checking out and testing this PR.  When I try to run it in the old picard-style without arguments (which should give a list of the tools and their program groups) I get the following error:

Exception in thread "main" java.lang.RuntimeException: The following classes are missing the required CommandLineProgramProperties annotation: CollectWgsMetricsFromQuerySorted, DummyProgram
	at picard.cmdline.PicardCommandLine.extractCommandLineProgram(PicardCommandLine.java:132)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:93)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:108)

**Comment by cmnbroad** [comment]
@ktibbett Hi Kathleen - I'm not sure what directions you're referring to, but CollectWgsMetricsFromQuerySorted has been removed from the Picard repo for a while now (I have no idea what DummyProgram is). Are you sure you did a clean build from this branch ? If I pull this PR and build it, and just run the jar, I get the tool/program group list as expected.

**Comment by vdauwera** [comment]
@ktibbett all basic invocations work for me -- @ktibbett, can you post the command you ran?

**Comment by vdauwera** [comment]
@nh13 @tfenne Would you care to review this? This PR affects how tool docs are categorized; the main driver was to standardize doc categories across Picard and GATK ahead of the GATK 4.0 release, which will produce more user-friendly documentation fro both projects. Shouldn't affect anything on the dev side, except if you have any open branches you'll  need to rebase to take in the new program group definitions.

We're hoping to get this merged ASAP once everything checks out in order to finalize preparations on the GATK side, so we would need a quick turnaround. Thanks!

**Comment by sooheelee** [comment]
@ktibbett, I don't know what `old picard-style without arguments` refers to. Can you please clarify? 

I think the following will be helpful to you.

I built a jar off of this branch and if I run `java -jar build/libs/picard.jar -h`, I see an index of tools that reflect the new categorization, e.g.: 

screenshot 2018-01-04 12 19 34

I am able to look at tool-specific documentation, e.g. with `java -jar build/libs/picard.jar FifoBuffer --help`:

screenshot 2018-01-04 12 22 38

Sidenote for @cmnbroad: If I run `--list` instead of `-h` I get the same tool list and a funny message at the end:

> '--list' is not a valid command. See PicardCommandLine -h for more information.

**Comment by ktibbett** [comment]
@vdauwera / @sooheelee , I did a clean checkout and this time it worked; apologies.

**Comment by sooheelee** [comment]
We've waited till EOD for input from @nh13 @tfenne and so will proceed assuming this is all agreeable with them. Since this PR is a blocker for many other Picard program group assignments, I will go ahead and merge.

**Comment by sooheelee** [comment]
Looks like these are due to other doc update PRs that have since been merged. Sorry about that @cmnbroad. Hope you find them easy to resolve.

**Comment by nh13** [comment]
Sorry but I am on vacation so I won’t be able to review. The amount of documentation changes is a bit overwhelming to follow, especially over the holidays! The changes also differ significantly from how I write documentation (not a bad thing, just different), so I will pass on reviewing any documentation changes and focus on functional changes moving forward. Thanks for pushing on all these PRs and including me.

**Pull request merged**